RecoEgamma/EgammaTools/plugins/HGCalElectronIDValueMapProducer.cc

```diff
@@ -123,35 +124,18 @@ HGCalElectronIDValueMapProducer::produce(edm::Event& iEvent, const edm::EventSet
       // Fill here all the ValueMaps from their appropriate functions
 
       // Energies / PT
-      maps_["gsfTrackPt"].push_back(electron.gsfTrack()->pt());
-      maps_["pOutPt"].push_back(std::sqrt(electron.trackMomentumAtVtx().perp2()));
-      maps_["scEt"].push_back(electron.superCluster()->energy() / std::cosh(electron.superCluster()->eta()));
-      maps_["scEnergy"].push_back(electron.superCluster()->energy());
-      maps_["ecOrigEt"].push_back(electron.electronCluster()->energy() / std::cosh(electron.electronCluster()->eta()));
-      maps_["ecOrigEnergy"].push_back(electron.electronCluster()->energy());
+      const auto* eleCluster = electron.electronCluster().get();
+      maps_["ecOrigEt"].push_back(eleCluster->energy() / std::cosh(eleCluster->eta()));
```

Comment:
```eleCluster->position().rho()/eleCluster->position().r()``` is numerically better and faster

Reply:
same for all other cases of ```/std::cosh(eleCluster->eta())```

Reply:
Just using the common calculation of Et: https://github.com/cms-sw/cmssw/search?utf8=%E2%9C%93&q=cosh+eta+energy&type=